src/sentry/lang/native/symbolicator.py

```diff
@@ -61,14 +61,7 @@
         "appconnectPrivateKey": {"type": "string"},
         "appName": {"type": "string", "minLength": 1, "maxLength": 512},
         "appId": {"type": "string", "minLength": 1},
-        # TODO(itunes): All of the below fields are deprecated. Remove together with migration.
-        "itunesUser": {"type": "string", "minLength": 1, "maxLength": 100},
-        "itunesCreated": {"type": "string", "format": "date-time"},
-        "itunesPassword": {"type": "string"},
-        "itunesSession": {"type": "string"},
         "bundleId": {"type": "string", "minLength": 1},
-        "orgPublicId": {"type": "string", "minLength": 36, "maxLength": 36},
-        "orgName": {"type": "string", "minLength": 1, "maxLength": 512},
```

Comment:
I don't think we can delete these right away and think multiple PRs are needed to achieve this:
1. Make these fields optional, stop writing them. Since the migration is not run automatically this can include the migration
2. Run the migration sometime.
3. Finally remove the fields from the schema

I'm not really sure how this works with on-premise...

Reply:
Agreed, since the migration is marked as `is_dangerous`, it'll run after the deploy, so removing the fields should be a separate PR.

Adding to the above: For on-premise deploys, we define to stop ingest consumers and celery workers to run the migrations. Same applies to single tenant (ST) deploys. It is accepted that the UI may temporarily glitch during these updates.

This means that you do not have to special-case for on-premise and ST beyond SaaS.

Reply:
thanks for the info on how on-prem is handled for migrations; it's a relief to hear that we don't have to special case for that configuration.

i've split this PR up into 3 (i was aiming for 2 but i ended up creating 3 anyways as @flub suggested):
1. https://github.com/getsentry/sentry/pull/29863 which marks the credentials optional and stops writing them
2. this PR, which wipes all existing configs via a migration
3. https://github.com/getsentry/sentry/pull/29865, which actually removes the fields from the schema